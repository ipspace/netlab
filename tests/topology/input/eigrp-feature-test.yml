#
# Sample multi-vendor configuration that test most features supported by OSPF configuration module
#
# Used in manual testing (feel free to write an automated test script ;)
#
addressing:
  passive:
    ipv4: 172.31.0.0/16
    prefix: 24
    ipv6: 2008:db8:0001::/48

module: [ eigrp ]

nodes:
  c_nxos:
    device: nxos
    eigrp.af.ipv4: True
    eigrp.af.ipv6: False
    eigrp.as: 123
  c_csr:
    device: csr
  c_ios:
    device: iosv

links:
- c_nxos-c_ios
- c_ios-c_csr

# Stub links:
- role: passive
  c_nxos:
  c_ios:
  c_csr:

- c_nxos:
- c_csr:
- c_ios:

# External link - should be passive
- role: external
  c_nxos:
  c_ios:
  c_csr:
