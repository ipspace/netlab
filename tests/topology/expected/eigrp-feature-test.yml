eigrp:
  as: 1
input:
- topology/input/eigrp-feature-test.yml
- package:topology-defaults.yml
links:
- interfaces:
  - ifindex: 1
    ifname: Ethernet1/1
    ipv4: 10.1.0.2/30
    node: c_nxos
  - ifindex: 1
    ifname: GigabitEthernet0/1
    ipv4: 10.1.0.1/30
    node: c_ios
  linkindex: 1
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  type: p2p
- interfaces:
  - ifindex: 2
    ifname: GigabitEthernet0/2
    ipv4: 10.1.0.6/30
    node: c_ios
  - ifindex: 2
    ifname: GigabitEthernet2
    ipv4: 10.1.0.5/30
    node: c_csr
  linkindex: 2
  node_count: 2
  prefix:
    ipv4: 10.1.0.4/30
  type: p2p
- bridge: input_3
  interfaces:
  - ifindex: 2
    ifname: Ethernet1/2
    ipv4: 172.31.0.1/24
    ipv6: 2008:db8:1::1/64
    node: c_nxos
  - ifindex: 3
    ifname: GigabitEthernet0/3
    ipv4: 172.31.0.3/24
    ipv6: 2008:db8:1::3/64
    node: c_ios
  - ifindex: 3
    ifname: GigabitEthernet3
    ipv4: 172.31.0.2/24
    ipv6: 2008:db8:1::2/64
    node: c_csr
  linkindex: 3
  node_count: 3
  prefix:
    ipv4: 172.31.0.0/24
    ipv6: 2008:db8:1::/64
  role: stub
  type: lan
- bridge: input_4
  interfaces:
  - ifindex: 3
    ifname: Ethernet1/3
    ipv4: 172.16.0.1/24
    node: c_nxos
  linkindex: 4
  node_count: 1
  prefix:
    ipv4: 172.16.0.0/24
  type: stub
- bridge: input_5
  interfaces:
  - ifindex: 4
    ifname: GigabitEthernet4
    ipv4: 172.16.1.2/24
    node: c_csr
  linkindex: 5
  node_count: 1
  prefix:
    ipv4: 172.16.1.0/24
  type: stub
- bridge: input_6
  interfaces:
  - ifindex: 4
    ifname: GigabitEthernet0/4
    ipv4: 172.16.2.3/24
    node: c_ios
  linkindex: 6
  node_count: 1
  prefix:
    ipv4: 172.16.2.0/24
  type: stub
- bridge: input_7
  interfaces:
  - ifindex: 4
    ifname: Ethernet1/4
    ipv4: 172.16.3.1/24
    node: c_nxos
  - ifindex: 5
    ifname: GigabitEthernet0/5
    ipv4: 172.16.3.3/24
    node: c_ios
  - ifindex: 5
    ifname: GigabitEthernet5
    ipv4: 172.16.3.2/24
    node: c_csr
  linkindex: 7
  node_count: 3
  prefix:
    ipv4: 172.16.3.0/24
  role: external
  type: lan
module:
- eigrp
name: input
nodes:
  c_csr:
    af:
      ipv4: true
      ipv6: true
    box: cisco/csr1000v
    device: csr
    eigrp:
      af:
        ipv4: true
        ipv6: true
      as: 1
      router_id: 10.0.0.2
    id: 2
    interfaces:
    - eigrp:
        passive: false
      ifindex: 2
      ifname: GigabitEthernet2
      ipv4: 10.1.0.5/30
      linkindex: 2
      name: c_csr -> c_ios
      neighbors:
      - ifname: GigabitEthernet0/2
        ipv4: 10.1.0.6/30
        node: c_ios
      type: p2p
    - bridge: input_3
      eigrp:
        passive: true
      ifindex: 3
      ifname: GigabitEthernet3
      ipv4: 172.31.0.2/24
      ipv6: 2008:db8:1::2/64
      linkindex: 3
      name: c_csr -> [c_nxos,c_ios]
      neighbors:
      - ifname: Ethernet1/2
        ipv4: 172.31.0.1/24
        ipv6: 2008:db8:1::1/64
        node: c_nxos
      - ifname: GigabitEthernet0/3
        ipv4: 172.31.0.3/24
        ipv6: 2008:db8:1::3/64
        node: c_ios
      role: stub
      type: lan
    - bridge: input_5
      eigrp:
        passive: true
      ifindex: 4
      ifname: GigabitEthernet4
      ipv4: 172.16.1.2/24
      linkindex: 5
      name: c_csr -> stub
      neighbors: []
      type: stub
    - bridge: input_7
      ifindex: 5
      ifname: GigabitEthernet5
      ipv4: 172.16.3.2/24
      linkindex: 7
      name: c_csr -> [c_nxos,c_ios]
      neighbors:
      - ifname: Ethernet1/4
        ipv4: 172.16.3.1/24
        node: c_nxos
      - ifname: GigabitEthernet0/5
        ipv4: 172.16.3.3/24
        node: c_ios
      role: external
      type: lan
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.2/32
      neighbors: []
      type: loopback
    mgmt:
      ifname: GigabitEthernet1
      ipv4: 192.168.121.102
      mac: 08:4f:a9:00:00:02
    min_mtu: 1500
    module:
    - eigrp
    name: c_csr
  c_ios:
    af:
      ipv4: true
      ipv6: true
    box: cisco/iosv
    device: iosv
    eigrp:
      af:
        ipv4: true
        ipv6: true
      as: 1
      router_id: 10.0.0.3
    id: 3
    interfaces:
    - eigrp:
        passive: false
      ifindex: 1
      ifname: GigabitEthernet0/1
      ipv4: 10.1.0.1/30
      linkindex: 1
      name: c_ios -> c_nxos
      neighbors:
      - ifname: Ethernet1/1
        ipv4: 10.1.0.2/30
        node: c_nxos
      type: p2p
    - eigrp:
        passive: false
      ifindex: 2
      ifname: GigabitEthernet0/2
      ipv4: 10.1.0.6/30
      linkindex: 2
      name: c_ios -> c_csr
      neighbors:
      - ifname: GigabitEthernet2
        ipv4: 10.1.0.5/30
        node: c_csr
      type: p2p
    - bridge: input_3
      eigrp:
        passive: true
      ifindex: 3
      ifname: GigabitEthernet0/3
      ipv4: 172.31.0.3/24
      ipv6: 2008:db8:1::3/64
      linkindex: 3
      name: c_ios -> [c_nxos,c_csr]
      neighbors:
      - ifname: Ethernet1/2
        ipv4: 172.31.0.1/24
        ipv6: 2008:db8:1::1/64
        node: c_nxos
      - ifname: GigabitEthernet3
        ipv4: 172.31.0.2/24
        ipv6: 2008:db8:1::2/64
        node: c_csr
      role: stub
      type: lan
    - bridge: input_6
      eigrp:
        passive: true
      ifindex: 4
      ifname: GigabitEthernet0/4
      ipv4: 172.16.2.3/24
      linkindex: 6
      name: c_ios -> stub
      neighbors: []
      type: stub
    - bridge: input_7
      ifindex: 5
      ifname: GigabitEthernet0/5
      ipv4: 172.16.3.3/24
      linkindex: 7
      name: c_ios -> [c_nxos,c_csr]
      neighbors:
      - ifname: Ethernet1/4
        ipv4: 172.16.3.1/24
        node: c_nxos
      - ifname: GigabitEthernet5
        ipv4: 172.16.3.2/24
        node: c_csr
      role: external
      type: lan
    loopback:
      ifindex: 0
      ifname: Loopback0
      ipv4: 10.0.0.3/32
      neighbors: []
      type: loopback
    mgmt:
      ifname: GigabitEthernet0/0
      ipv4: 192.168.121.103
      mac: 08:4f:a9:00:00:03
    module:
    - eigrp
    name: c_ios
  c_nxos:
    af:
      ipv4: true
      ipv6: true
    box: cisco/nexus9300v
    device: nxos
    eigrp:
      af:
        ipv4: true
      as: 123
      router_id: 10.0.0.1
    id: 1
    interfaces:
    - eigrp:
        passive: false
      ifindex: 1
      ifname: Ethernet1/1
      ipv4: 10.1.0.2/30
      linkindex: 1
      name: c_nxos -> c_ios
      neighbors:
      - ifname: GigabitEthernet0/1
        ipv4: 10.1.0.1/30
        node: c_ios
      type: p2p
    - bridge: input_3
      eigrp:
        passive: true
      ifindex: 2
      ifname: Ethernet1/2
      ipv4: 172.31.0.1/24
      ipv6: 2008:db8:1::1/64
      linkindex: 3
      name: c_nxos -> [c_ios,c_csr]
      neighbors:
      - ifname: GigabitEthernet0/3
        ipv4: 172.31.0.3/24
        ipv6: 2008:db8:1::3/64
        node: c_ios
      - ifname: GigabitEthernet3
        ipv4: 172.31.0.2/24
        ipv6: 2008:db8:1::2/64
        node: c_csr
      role: stub
      type: lan
    - bridge: input_4
      eigrp:
        passive: true
      ifindex: 3
      ifname: Ethernet1/3
      ipv4: 172.16.0.1/24
      linkindex: 4
      name: c_nxos -> stub
      neighbors: []
      type: stub
    - bridge: input_7
      ifindex: 4
      ifname: Ethernet1/4
      ipv4: 172.16.3.1/24
      linkindex: 7
      name: c_nxos -> [c_ios,c_csr]
      neighbors:
      - ifname: GigabitEthernet0/5
        ipv4: 172.16.3.3/24
        node: c_ios
      - ifname: GigabitEthernet5
        ipv4: 172.16.3.2/24
        node: c_csr
      role: external
      type: lan
    loopback:
      ifindex: 0
      ifname: loopback0
      ipv4: 10.0.0.1/32
      neighbors: []
      type: loopback
    mgmt:
      ifname: mgmt0
      ipv4: 192.168.121.101
      mac: 08:4f:a9:00:00:01
    module:
    - eigrp
    name: c_nxos
provider: libvirt
